File: 0101-symmetric-tree/0101-symmetric-tree.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func checkMirrored(n1, n2 *TreeNode) bool {
    if n1 == nil && n2 == nil {
        return true
    }
    
    if n1 == nil || n2 == nil {
        return false
    }
    
    if n1.Val != n2.Val {
        return false
    }
    
    return checkMirrored(n1.Left, n2.Right) && checkMirrored(n1.Right, n2.Left)
}

func isSymmetric(root *TreeNode) bool {
    if root == nil {
        return true
    }
    
    return checkMirrored(root.Left, root.Right)
}

File: 0101-symmetric-tree/0101-symmetric-tree.py
def checkMirrored(node1: Optional[TreeNode], node2: Optional[TreeNode]) -> bool:
    if node1 == None and node2 == None:
        return True
    
    if node1 == None or node2 == None:
        return False
    
    if node1.val != node2.val:
        return False
    
    return checkMirrored(node1.left, node2.right) and checkMirrored(node1.right, node2.left)

# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def isSymmetric(self, root: Optional[TreeNode]) -> bool:
        return checkMirrored(root.left, root.right)